src/bid/bid.rs

```diff
@@ -60,6 +59,81 @@ impl Borrow<u64> for Bid {
     }
 }
 
+impl PartialEq for Bid {
+    fn eq(&self, other: &Self) -> bool {
+        self.hash().eq(&other.hash())
+    }
+}
+
+// This needs to be between braces since const fn calls passed as const_generics
+// params aren't perfectly supported yet.
+impl Serializable<{ Bid::serialized_size() }> for Bid {
+    type Error = BlindBidError;
+
+    fn from_bytes(buf: &[u8; Self::SIZE]) -> Result<Bid, Self::Error> {
+        let mut one_cipher = [0u8; PoseidonCipher::cipher_size_bytes()];
+        let mut one_scalar = [0u8; 32];
+        let mut one_stealth_address = [0u8; 64];
+        let mut one_u64 = [0u8; 8];
+
+        one_cipher[..]
+            .copy_from_slice(&buf[0..PoseidonCipher::cipher_size_bytes()]);
+        let encrypted_data = PoseidonCipher::from_bytes(&one_cipher)?;
+
+        one_scalar[..]
+            .copy_from_slice(&buf[PoseidonCipher::cipher_size_bytes()..128]);
+        let nonce = BlsScalar::from_bytes(&one_scalar)?;
```

Comment:
For the type that supports `Serializable` you don't need to do that anymore.

You can directly write:
```rust
let encrypted_data = PoseidonCipher::from_slice(&buf);
let nonce = BlsScalar::from_slice(&buf[PoseidonCipher::SIZE..];
```

etc.

However it would be probably better to implement the following issue: https://github.com/dusk-network/dusk-bytes/issues/11 or similar API once we find a good ones.